android/src/main/java/com/swmansion/rnscreens/CustomToolbar.kt

```diff
@@ -19,16 +21,22 @@ open class CustomToolbar(
             override fun doFrame(frameTimeNanos: Long) {
                 isLayoutEnqueued = false
                 measure(
-                    MeasureSpec.makeMeasureSpec(width, MeasureSpec.EXACTLY),
-                    MeasureSpec.makeMeasureSpec(height, MeasureSpec.EXACTLY),
+                    MeasureSpec.makeMeasureSpec(width, MeasureSpec.AT_MOST),
+                    MeasureSpec.makeMeasureSpec(height, MeasureSpec.AT_MOST),
                 )
                 layout(left, top, right, bottom)
             }
         }
 
     override fun requestLayout() {
         super.requestLayout()
-        if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.Q) {
+        val softInputMode =
+            (context as ThemedReactContext)
+                .currentActivity
+                ?.window
+                ?.attributes
+                ?.softInputMode
+        if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.Q && softInputMode == WindowManager.LayoutParams.SOFT_INPUT_ADJUST_PAN) {
```

Comment:
I like introduction of this check, however I need you to confirm that issue from #2332 is not reintroduced.

Reply:
From what I've tested the issue mentioned only appeared when mode was set to `adjustPan`. I am not 100% certain though. @tboba Do you remember if it was an issue with other modes?

Reply:
I was involved in fixing it & can confirm that it occurred only for `adjustPan`.

However have you tested that? You do set this in `AndroidManifest` file of the application under the `softInputMode` key (or something similar).

Reply:
I have tested it and didn't found any problems with softInputModes other than `adjustPan`, which is handled by this check.